### PR TITLE
Documentation improvements, handle non-lowercase file extensions, and add public method to scan from within Java

### PR DESCRIPTION
**Pull request by mica-alex**
This PR adds two commits, one to improve Main.java and one to improve Detector.java.

The commit for `Main.java` adds the following:
1. Javadocs and comments throughout
2. Improved console output by compiling a list of infected files to show at the end
3. Added support for running a scan from within Java (i.e. launcher automatically check downloaded mod files, etc.)
4. Converted file name to lowercase when checking file extension to ensure capitalization does not affect scanning (i.e. JAR, JaR, JAr, jaR, etc.)

The commit for `Detector.java` adds the following:
1. Javadocs and comments for the class itself and the scan method.
2. Improved the scan method so it returns a boolean indicating if infection was found or not.
3. Converted file name to lowercase when checking for .class file extension to ensure capitalization does not affect scanning (i.e. CLASS, claSS, CLasS, etc.)

### CODE REVIEW

**Comment by mica-alex** [comment]
Will fix conflicts and update

**Comment by mica-alex** [comment]
@MCRcortex @IMS212 @Huskydog9988 This pull request is ready for review now. My apologies for tagging you if not the correct person, but I just wanted to make sure this was back on the radar now that earlier conflicts are resolved.

**Comment by Huskydog9988** [comment]
we are currently working on some issues with detection rn, but you have the clear from me

**Comment by mica-alex** [comment]
> we are currently working on some issues with detection rn, but you have the clear from me

Sounds good. Hopefully everyone approves/like these changes., and let me know if you need anything with regards to changes.

File: src/main/java/me/cortex/jarscanner/Gui.java [hunk 1]
```diff
@@ -45,15 +46,14 @@ private static void createAndDisplayGui() {
 
         JButton runButton = new JButton("Run!");
         runButton.addActionListener(e -> {
-            Main.run(4, searchDir, true, out -> {
-                textArea.append(out + "\n");
-                return out;
-            });
-
-            Detector.checkForStage2(s -> {
```
Comment (mica-alex) [inline src/main/java/me/cortex/jarscanner/Gui.java:53]: Also, it looks scary to be removing the stage 2 check here, but that is already called in Main.run(), so there is no need to run it twice.

**Comment by mica-alex** [comment]
Eating dinner then will fix conflicts

**Comment by Huskydog9988** [comment]
Hey are you fine reimplementing the color codes from #23? Currently we may need to drop them temporally because the gui can't render the ansi colors.

**Comment by mica-alex** [comment]
> Hey are you fine reimplementing the color codes from #23? Currently we may need to drop them temporally because the gui can't render the ansi colors.

Sorry I did not see this sooner. I have that handled already in this PR. I do a replaceAll on the ANSI color codes which I moved in to a constants file

**Comment by mica-alex** [comment]
Oh, I did not see that more PRs were merged. More conflicts....again....

**Comment by mica-alex** [comment]
Ok, I have addressed all conflicts

**Comment by mica-alex** [comment]
Fixes #27

**Comment by mica-alex** [comment]
Corrected a mistake (sorry) where I omitted output of compiled scan results on the GUI.

**Comment by Huskydog9988** [comment]
our discord is now public, you can coordinate with us here if you want  https://discord.gg/zPdFK47682 (we are currently working on a rewite)

**Comment by mica-alex** [comment]
I will try joining in the morning. 

I am fairly discouraged. 😞 At this point, I have had to handle merge conflicts 4+ times, some rather large.

Hopefully there is a review soon, so these changes can be rebased into the rewrite branch and not lost again. There is a great deal of improvement here that shouldn't be missed including:

1. Better Javadocs
2. Documentation fixes
3. Main class improvements to allow launchers, etc. to start a scan and receive a list of results (infected files)
4. ANSI color code handling
5. Auto-scroll for log in GUI (there is a checkbox to turn it off too)
6. Cancel button in GUI
7. more...

**Comment by Huskydog9988** [comment]
yea hopefully we can get one in, ill try and remind the maintainers tomorrow

**Comment by MCRcortex** [comment]
sorr looking at it now

**Pull request merged**